modules/programs/ssh.nix

```diff
@@ -518,14 +535,14 @@ in {
 
     home.packages = optional (cfg.package != null) cfg.package;
 
-    home.file.".ssh/config".text = let
+    home.file.".ssh/config".source = mkIf cfg.internallyManaged cfg.configPath;
```

Comment:
If you just have `programs.ssh.enable = true;` with no further tweaks in a `home-manager.users.<user>` block, this change causes the build to blow up with

```
error: The option `home-manager.users.<user>.home.file.".ssh/config".source' was accessed but has no value defined. Try setting the option.
```

Reply:
You need to change this to wrap the `source` setting in the mkIf instead of calling mkIf on source itself.